test/multi_process_test.rb

```diff
@@ -28,10 +28,89 @@
     Resque.schedule = schedules
     counts << Resque.schedule.size
 
-    threads.each(&:join)
+    counts += get_results_from_children(pids)
 
     counts.each_with_index do |c, i|
       assert_equal schedule_count, c, "schedule count is incorrect (c: #{i})"
     end
   end
+
+  test 'concurrent shutdowns and startups do not corrupt the schedule' do
+    counts = []
+    children = []
+
+    processes = 40
+
+    schedules = {}
+    schedule_count = 300
+    schedule_count.times do |n|
+      schedules["job_#{n}"] = { 'cron' => '0 1 0 0 0' }
+    end
+
+    Resque.schedule = schedules
+
+    processes.times do |n|
+      children << fork_with_marshalled_pipe_and_result do
```

Comment:
Yeah this is an interesting point, from https://github.com/resque/resque-scheduler/pull/439#discussion_r16788812:
>Because we're using the same redis connection for all the threads, we're actually bound on the Redis#synchronize mutex here and not testing the actual cross-process concurrency (which would also explain why Redis#pipelined was causing this test to succeed). Maybe we need to fork to test this one?